fish-rust/src/builtins/mod.rs

```diff
@@ -28,3 +28,20 @@ pub mod wait;
 
 // Note these tests will NOT run with cfg(test).
 mod tests;
+
+mod prelude {
+    pub use super::shared::*;
+    pub use libc::c_int;
+
+    pub(crate) use crate::{
+        ffi::{self, parser_t, separation_type_t, Repin},
+        wchar::prelude::*,
+        wchar_ffi::{c_str, AsWstr, WCharFromFFI, WCharToFFI},
+        wgetopt::{
+            wgetopter_t, wopt, woption,
+            woption_argument_t::{self, *},
+            NONOPTION_CHAR_CODE,
+        },
+        wutil::{fish_iswalnum, fish_wcstoi, fish_wcstol, fish_wcstoul},
```

Comment:
Nit: I'm not sure where to draw the line on what to include.
Some of these are not specific to builtins but I guess that's fine for the FFi types.
That leaves `fish_iswalnum` etc. which could be in another prelude.. but we can fix that later.

Reply:
If we compare this to plain Rust, this is mostly handled by `str::parse`, and therefore this is "by default available", which is why I included them. They are all namespaces behind `fish_` so including them should not be problematic. 

I guess `fish_is_walnum` is an odd-one out in that regard, and does not follow the above rationale. I will remove it.

I will also split out the prelude-adding and the adoption, sorry for making it hard to find the actual changes!